wolfcrypt/src/asn.c

```diff
@@ -5111,13 +5122,23 @@ int wc_DsaParamsDecode(const byte* input, word32* inOutIdx, DsaKey* key,
 int DsaPrivateKeyDecode(const byte* input, word32* inOutIdx, DsaKey* key,
                         word32 inSz)
 {
-    int    length, version, ret = 0, temp = 0;
+    int length, version, ret = 0, temp = 0;
+#if defined(HAVE_PKCS8) || defined(HAVE_PKCS12)
+    word32 algId = 0;
+#endif
 
     /* Sanity checks on input */
     if (input == NULL || inOutIdx == NULL || key == NULL) {
         return BAD_FUNC_ARG;
     }
 
+#if defined(HAVE_PKCS8) || defined(HAVE_PKCS12)
+    /* if has pkcs8 header skip it */
+    if (ToTraditionalInline_ex(input, inOutIdx, inSz, &algId) < 0) {
```

Comment:
just calling ToTraditioanlInline_ex not enough to skip header?

Reply:
The inOutIdx handles skipping

Reply:
Needs If statement?

Reply:
No, the `inOutIdx` is updated only on success. On failure there is no PKCS8 header and it uses the existing flow. This just tries to parse a PKCS8 header with ToTraditionalInline_ex and if it is found sets the `inOutIdx` to skip it. The original buffer is not altered, just the index to it.

Reply:
Ok, thanks